common/corerequestcontext.go

```diff
@@ -88,17 +88,24 @@ func UpdateResponseStatus(ctx context.Context, status int) error {
 	return nil
 }
 
-func CoreRequestContextMiddleware(logger ...*logrus.Logger) func(next http.Handler) http.Handler {
+func CoreRequestContextMiddleware(logger *logrus.Logger) func(next http.Handler) http.Handler {
+	ctx := LoggerToContext(context.Background(), logger, nil)
+	return CoreRequestContextMiddlewareWithContext(ctx)
+}
+
+func CoreRequestContextMiddlewareWithContext(ctx context.Context) func(next http.Handler) http.Handler {
 	var ctxlogger *logrus.Logger
-	if len(logger) == 1 {
-		ctxlogger = logger[0]
+	if getCoreContext(ctx) != nil {
+		ctxlogger = getCoreContext(ctx).logger
 	}
 	return func(next http.Handler) http.Handler {
 		fn := func(w http.ResponseWriter, r *http.Request) {
 			ctx := r.Context()
 			if ctxlogger != nil {
 				ctx = LoggerToContext(ctx, ctxlogger, nil)
 			}
+			logger := log.NewStandardLogger()
+			ctx = log.WithLogger(logger).Onto(ctx)
```

Comment:
@andrewemeryanz Can't get pkg logger from the ctx passed in (no key to log.WithLogger) , so I'm using this approach. Not sure if this is the right way?

Reply:
Use `log.From(ctx)` to retrieve the logger. Cache the logger outside of the `return func` to avoid it being executed on each call.